### PR TITLE
Noise detection, slow taxi, and other tests for ground methods

### PR DESCRIPTION
**Pull request by RaphaelDELAIR**
Where should I put tests ? How should I create a dataset for flights that are used to debug and assess results ?

### CODE REVIEW

File: traffic/algorithms/navigation.py [hunk 1]
```diff
@@ -942,7 +957,120 @@ def pushback(
...
+        self = cast("Flight", self)
+
+        duration_threshold = to_timedelta(min_duration)
+
+        current_flight = self.moving()
+        if current_flight is None:
             return None
 
-        return in_movement.before(first_backwards.start)
+        current_flight = current_flight.onground()
```
Comment (xoolive) [inline traffic/algorithms/navigation.py:1015]: do you really need that? if yes, why not before "moving()"?

**Review by xoolive — CHANGES_REQUESTED** [review]
Thanks a lot for the propositions! Let's finish it, include tests and we should be done!

File: traffic/algorithms/navigation.py [hunk 2]
```diff
@@ -942,7 +957,120 @@ def pushback(
...
+        # TODO focus on ground trajectories
+        if "compute_track" not in self.data.columns:
+            self = self.cumulative_distance(compute_gs=False)
+
+        freq = (
+            self.diff("compute_track")
+            .compute_track_diff.round()
+            .value_counts(normalize=True)
+        )
+        if 90 not in freq.index or -90 not in freq.index:
```
[the remark this answers is not in the document]
Comment (RaphaelDELAIR) [reply traffic/algorithms/navigation.py:992]: As it is a normalized value_counts, it computes the frequency is the number of 90 degrees turns divided by the number of data point of the trajectory, in that sense, those turns do not depend on the time but more on the amount of row of our trajectory

File: traffic/core/traffic.py [hunk 3]
```diff
@@ -743,6 +753,57 @@ def geoencode(self, *args, **kwargs):
         """
         raise NotImplementedError
 
+    def geoencode_groundtraffic(
```
Comment (xoolive) [inline traffic/core/traffic.py:756]: I will reject this part for this PR, but keep the suggestion for another increment

**Pull request merged**